File: facade/java/FacadeTest.java
public class FacadeTest {
    public static void main(String[] args) {
        String phone = "13800138000";
        String vcode = "123321";

        Facade facade = new Facade();
        boolean flag = facade.loginOrRegister(phone, vcode);
        if (flag) {
            System.out.println("【登录或注册】成功");
        } else {
            System.err.println("【登录或注册】失败");
        }

        if (facade.userService().login(phone, vcode)) {
            System.out.println("【登录或注册】重新登录成功");
        } else {
            System.err.println("【登录或注册】重新登录失败");
        }
    }
}

File: facade/go/facade_test.go
package facade

import "testing"

func TestFacade(t *testing.T) {
	phone, vcode := "13800138000", "123321"

	f := NewFacade()
	if ok, err := f.LoginOrRegister(phone, vcode); ok && err == nil {
		t.Log("【登录或注册】成功")
	} else {
		t.Error("【登录或注册】失败")
	}

	if ok, err := f.UserService().Login(phone, vcode); ok && err == nil {
		t.Log("【登录或注册】重新登录成功")
	} else {
		t.Error("【登录或注册】重新登录失败")
	}
}
